packages/flutter_tools/test/integration/debugger_stepping_test.dart

```diff
@@ -0,0 +1,51 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/file.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+
+import '../src/common.dart';
+import 'test_data/stepping_project.dart';
+import 'test_driver.dart';
+import 'test_utils.dart';
+
+void main() {
+  group('debugger', () {
+    Directory tempDir;
+    final SteppingProject _project = SteppingProject();
+    FlutterRunTestDriver _flutter;
+
+    setUp(() async {
+      tempDir = createResolvedTempDirectorySync();
+      await _project.setUpIn(tempDir);
+      _flutter = FlutterRunTestDriver(tempDir);
+    });
+
+    tearDown(() async {
+      await _flutter.stop();
+      tryToDelete(tempDir);
+    });
+
+    test('can step over statements', () async {
+      await _flutter.run(withDebugger: true);
+
+      // Stop at the initial breakpoint that the expected steps are based on.
+      await _flutter.breakAt(_project.breakpointUri, _project.breakpointLine, restart: true);
```

Comment:
i'm changing this in my PR to start the app paused, set the breakpoint, resume, then wait for it to hit the breakpoint. This avoids the need for the restart.

Reply:
Sounds reasonable to me :-)